src/main/java/Duke.java

```diff
@@ -1,10 +1,92 @@
+package seedu.duke;
+
+import seedu.duke.Ui;
+import seedu.duke.Storage;
+import seedu.duke.Parser;
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.DukeNoSuchCommandException;
+
+/**
+ * Main class of this project.
+ * Personal assistant bot to help you keep track of tasks.
+ */
+
 public class Duke {
-    public static void main(String[] args) {
+
+    private Ui ui;
+    private Parser parser;
+    private TaskList taskList;
+    private Storage storage;
+
+    public Duke() {
+
+        ui = new Ui();
+        storage = new Storage("save.txt");
+        this.taskList = storage.load();
+
         String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+                                + "|  _ \\ _   _| | _____ \n"
+                                + "| | | | | | | |/ / _ \\\n"
+                                + "| |_| | |_| |   <  __/\n"
+                                + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        Ui.printBlock(new String[] {"Hello! I'm Duke.", "What can I do for you?"});
+
+        // Initializes the parser so that the loop runs at least once
+        try {
+            this.parser = new Parser("list");
+        } catch (Exception e) {
+            Ui.printBlock("Unexpected Error");
+        }
+
+    }
+
+    public void run() {
+
+        while (!parser.getType().equals("bye")) {
```

Comment:
I think this switch case is repeated in your Parser class, you might want to run the logic using the Parser class instead. (By creating the Command classes and passing the tasklist into the Commands' methods. 

Eg. 
In the parser:
public Command parse(String input) {
    switch (input):
    ...
    case "bye":
        return new ByeCommand;
    }
    ...
}

In command class:
public void execute(TaskList tasks, Ui ui, Storage storage) {
    ...
}

In duke:
Command c = parser.parse(input);
c.execute(tasks, ui, storage);

Reply:
Thanks for the input. I'll begin refactoring once I begin creating command classes.